docs/logs/README.md

```diff
@@ -222,6 +227,110 @@ code is now depending on which logger is being enabled, not to mention the
 argument evaluation might have significant side effects that are now depending
 on the logging configuration.
 
+:heavy_check_mark: You should use a dedicated parameter to log exceptions when
+using the compile-time source generator.
+
+```csharp
+var food = "tomato";
+var price = 2.99;
+
+try
+{
+    // Execute some logic
+
+    logger.SayHello(food, price);
+}
+catch (Exception ex)
+{
+    logger.SayHelloFailure(ex, food, price);
+}
+
+internal static partial class LoggerExtensions
+{
+    [LoggerMessage(Level = LogLevel.Information, Message = "Hello from {food} {price}.")]
+    public static partial void SayHello(this ILogger logger, string food, double price);
+
+    [LoggerMessage(Level = LogLevel.Error, Message = "Could not say hello from {food} {price}.")]
+    public static partial void SayHelloFailure(this ILogger logger, Exception exception, string food, double price);
+}
+```
+
+> [!NOTE]
+> When using the compile-time source generator the first `Exception` parameter
+> detected is automatically given special handling. It **SHOULD NOT** be part of
+> the message template.
+
+:heavy_check_mark: You should use the dedicated overloads to log exceptions when
+using the logging extensions methods.
```

Comment:
since we recommended against using extension methods in earlier section of the doc, lets not talk about it further :)

Reply:
That's a tough one. I don't disagree with you necessarily, but I see this mistake being made with users migrating to ILogger who are doing find/replace type of things (not taking the source generator). Perhaps we should soften the recommendation for source generator? It is great for high-perf. But users may not care and prefer the more concise syntax. There are also cases where high-perf probably doesn't matter. For example, startup/shutdown logging.

Reply:
Even .NET docs show both slow way and fast way, so we may not be able to completely get rid of extension methods from the docs, agree!